src/app/cli/src/commands/delete_command.rs

```diff
@@ -46,28 +48,24 @@ impl DeleteCommand {
 #[async_trait::async_trait(?Send)]
 impl Command for DeleteCommand {
     async fn run(&mut self) -> Result<(), CLIError> {
-        if self.dataset_refs.is_empty() && !self.all {
+        if self.dataset_ref_patterns.is_empty() && !self.all {
             return Err(CLIError::usage_error("Specify a dataset or use --all flag"));
         }
 
-        let dataset_refs = if self.all {
+        let dataset_refs: Vec<_> = if self.all {
             unimplemented!("Recursive deletion is not yet supported")
         } else if self.recursive {
             unimplemented!("Recursive deletion is not yet supported")
         } else {
-            self.dataset_refs.clone()
+            filter_datasets_by_pattern(
+                self.dataset_repo.as_ref(),
+                self.dataset_ref_patterns.clone(),
+            )
+            .map_ok(|dataset_handle| dataset_handle.as_local_ref())
+            .try_collect()
+            .await?
         };
 
-        // Check references exist
-        // TODO: PERF: Create a batch version of `resolve_dataset_ref`
-        for dataset_ref in &self.dataset_refs {
-            match self.dataset_repo.resolve_dataset_ref(dataset_ref).await {
-                Ok(_) => Ok(()),
-                Err(GetDatasetError::NotFound(e)) => Err(CLIError::usage_error_from(e)),
```

Comment:
Can you test `kamu delete dataset-name-that-doesnt-exist`? Previously `NotFound` error was translated into `CLIError::Usage` as you can see here, but now it looks like it will surface as internal error.